shuup/core/api/front_products.py

```diff
@@ -445,13 +438,14 @@ def best_selling(self, request):
 
         # take here the top `limit` records, because the filter_queryset below can mess with our work
         product_ids = [
-            d[0] for d in sorted(six.iteritems(combined_variation_products), key=lambda i: i[1], reverse=True)[:limit]
+            d[0] for d in sorted(six.iteritems(combined_variation_products), key=lambda i: i[1], reverse=True)
         ]
 
         shop_products_qs = ShopProduct.objects.filter(product__id__in=product_ids)
         shop_products_qs = self.filter_queryset(shop_products_qs).distinct()
+        shop_products_qs = self.paginate_queryset(shop_products_qs)
```

Comment:
Take a look here: https://github.com/encode/django-rest-framework/blob/master/rest_framework/mixins.py#L39
The `paginate_queryset` can return `None` if no page is needed